### DISCUSSION TITLE
Change domain of verdaccio

### DISCUSSION THREAD

**Discussion by REBELinBLUE** [verdaccio/verdaccio#4564] [Q&A]
Hi, this is possibly a silly question but I can't find anything related to this

I currently have Verdaccio installed via helm on `npm.foo.mydomain.com` but want to change it to `npm.mydomain.com`, I have changed the ingress host and VERDACCIO_PUBLIC_URL but I noticed that

`https://npm.mydomain.com/-/verdaccio/data/packages` contains a lot of references to the old domain in `publishConfig.registry`

Is this OK or is there something I should do to change it?

Similarly, I notice a lot of references when running ```find /verdaccio/storage/data/ -name "package.json" -exec grep -H "foo.mydomain.com" {} \;```, should these be changed?

**Comment by juanpicado** [comment]
> Similarly, I notice a lot of references when running find /verdaccio/storage/data/ -name "package.json" -exec grep -H "foo.mydomain.com" {} \;, should these be changed?

No, if you set `VERDACCIO_PUBLIC_URL` [the distribution file is being changed on every request](https://github.com/verdaccio/verdaccio/blob/master/packages/web/src/api/package.ts#L67). 

> https://npm.mydomain.com/-/verdaccio/data/packages contains a lot of references to the old domain in publishConfig.registry

🤔 I'll check what's that, could you provide examples (no real data)? just try to see maybe you found something.

**Reply by REBELinBLUE** [reply]
For example

```
[
  {
    "name": "@mydomain/addon-knobs",
    "version": "6.4.3",
    "description": "Storybook addon for editing props",
    "keywords": [
      "storybook-addons",
      "test"
    ],
    "repository": {
      "type": "git",
      "url": "git+https://github.com/storybookjs/addon-knobs.git"
    },
    "publishConfig": {
      "registry": "https://npm.foo.mydomain.com"
    },
    "license": "MIT",
    "main": "dist/index",
    "files": [
      "dist/**/*",
      "README.md",
      "*.js"
    ],
```

the dist tarball has been updated so maybe it is fine?

```
    "dist": {
      "integrity": "sha512-dpxZawGZ8aLnVaAl4HLDkfUp25MnzwLudAoPtEwMTz1UxKyEIBFQlzG07UIOSfwcpRLVn7S/KQZud7OX0l2sEA==",
      "shasum": "f0c35c2bc02ca1a0abf62b8a3e2004df08775fec",
      "tarball": "https://npm.mydomain.com/@mydomain/addon-knobs/-/addon-knobs-6.4.3.tgz"
    },
```

**Reply by cobolbaby** [reply]
By checking `npm i --verbose ...`, it appears that changing the domain name has no impact on package installation.

**Reply by juanpicado** [reply]
Tarballs dist url are being updated on the fly (peer request), so it is not persistent in the storage if you change the domain in the future.
